maths/fermats_little_theorem.dart

```diff
@@ -21,12 +19,22 @@ void main() {
   // a prime number
   int p = 701;
 
-  double a = 1000000000;
```

Comment:
Removing support for floating point numbers seems like a step too far.  Is there a way to keep supporting floating point numbers? Tests should cover positive numbers, negative numbers, and zero as well as floating point numbers if we keep supporting them.